src/main/java/application/AbstractBaseController.java

```diff
@@ -37,6 +56,70 @@
 
     private final Log log = LogFactory.getLog(AbstractBaseController.class);
 
+
+    public Object resolveFormGetNext(MenuSession menuSession) throws Exception {
+        menuSession.getSessionWrapper().syncState();
+        if(menuSession.getSessionWrapper().finishExecuteAndPop(menuSession.getSessionWrapper().getEvaluationContext())){
+            Object nextMenu = getNextMenu(menuSession);
+            menuSessionRepo.save(new SerializableMenuSession(menuSession));
+            return nextMenu;
+        }
+        return null;
+    }
+
+    public Object getNextMenu(MenuSession menuSession) throws Exception {
+        return getNextMenu(menuSession, 0);
+    }
+
+    private Object getNextMenu(MenuSession menuSession, int offset) throws Exception {
+        return getNextMenu(menuSession, offset, "");
+    }
+
+    protected Object getNextMenu(MenuSession menuSession, int offset, String searchText) throws Exception {
+
+        Screen nextScreen;
+
+        // If we were redrawing, remain on the current screen. Otherwise, advance to the next.
+        nextScreen = menuSession.getNextScreen();
+        // No next menu screen? Start form entry!
+        if (nextScreen == null) {
+            return generateFormEntryScreen(menuSession);
+        } else {
+            MenuBean menuResponseBean;
+
+            // We're looking at a module or form menu
+            if (nextScreen instanceof MenuScreen) {
+                menuResponseBean = generateMenuScreen((MenuScreen) nextScreen, menuSession.getSessionWrapper(),
+                        menuSession.getId());
+            }
+            // We're looking at a case list or detail screen (probably)
```

Comment:
Haha it actually can't be anything else at the moment, removing ambiguous comment.